### PR TITLE
Replace JS with MDX Doc

### LINKED ISSUE #25: Replace JS stories with MDX Docs
**Description**
Replace stories written in `.stories.js` files with MDX docs written in `.stories.mdx` format.

**Problems solved**
MDX presents a better way of combining code examples with long form documentation. It also enables the use of DocsPage components such as `Typeset` and `IconGallery`.

**Want to take up?**

[Yes|No]
Yes

### PR DESCRIPTION
**Pull request by megha070**
## Fixes

Fixes #25  by @megha070 

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by megha070** [comment]
image
@dhruvkb can you please help me with this error.

**Comment by dhruvkb** [comment]
@megha070 this occurs if you call a function within the same function, like a recursion without an exit condition.

**Comment by megha070** [comment]
@dhruvkb the decorators are the same for all the fonts except for the font-family, so as to remove repetitive code, I tried importing decorator by defining it in another file (story.js) but the contents are not visible. Could you please tell the reason behind this anomaly? 
This code is working fine as of now but why can't we import a decorator?

**Comment by dhruvkb** [comment]
You should ideally not import content from other stories. If two stories need some feature, add it to `helpers.js`.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
I've updated the **Source Sans Pro** story to indicate my expectations from an MDX perspective. Would be awesome if you could make the corresponding updates in the **Accidenz Commons** and **Roboto Condensed** stories.

**Comment by megha070** [comment]
@dhruvkb I couldn't think of using typeset in this. I would be extremely thankful if you could please direct me to some source that could help me with various components offered by @storybook/addon-blocks and their use-cases.

**Comment by megha070** [comment]
@dhruvkb Don't we have to include the sample text?

**Comment by dhruvkb** [comment]
That's the problem with Storybooks, it changes so quickly that all documentation quickly becomes very old. The best source of info on Storybooks is the code itself.

So if you go searching in
```
node_modules/@storybook/components/dist/blocks/index.d.ts
``` 
you'll find all the blocks they support.

File: src/stories/roboto_condensed.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  fontSizes={['2rem']}
+  fontFamily="Roboto Condensed"
+  fontWeight={700}
+  sampleText="0 1 2 3 4 5 6 7 8 9 + - * / % $" />
+
+<Typeset
+  fontSizes={['2rem']}
+  fontFamily="Roboto Condensed"
+  fontWeight={700}
+  sampleText=". ? ! ' & : ; @ # < > ( ) { } [ ]" />
```
Comment (dhruvkb) [inline src/stories/roboto_condensed.stories.mdx]: There are no semibold and bold variants in Roboto. Did you not see the `roboto_condensed.scss` file?

**Comment by megha070** [comment]
I see. Thanks a ton, will go through this once.

File: src/stories/roboto_condensed.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+</div>`
+  ]
+ }}/>
+
+# Roboto Condensed
+
+Roboto,a neo-grotesque sans-serif typeface family developed by Google,was designed by Christian Robertson. 
+It is a sans serif condensed typeface which is the default font on Android.
+
+It is used in Vocabulary in 3 font weights: regular (400), semibold (600) and bold (700).
```
Comment (dhruvkb) [inline src/stories/roboto_condensed.stories.mdx]: No. Just one: 400.

File: src/stories/roboto_condensed.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+</style>
+<div class="roboto-condensed has-text-black">
+  ${storyFn()}
+</div>`
+  ]
+ }}/>
+
+# Roboto Condensed
+
+Roboto,a neo-grotesque sans-serif typeface family developed by Google,was designed by Christian Robertson.
```
Comment (dhruvkb) [inline src/stories/roboto_condensed.stories.mdx]:
```suggestion
Roboto, a neo-grotesque sans-serif typeface family developed by Google, was designed by Christian Robertson. 
```

File: src/stories/roboto_condensed.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+<div class="roboto-condensed has-text-black">
+  ${storyFn()}
+</div>`
+  ]
+ }}/>
+
+# Roboto Condensed
+
+Roboto,a neo-grotesque sans-serif typeface family developed by Google,was designed by Christian Robertson. 
+It is a sans serif condensed typeface which is the default font on Android.
```
Comment (dhruvkb) [inline src/stories/roboto_condensed.stories.mdx]:
```suggestion
It is a sans-serif, condensed typeface which is the default font on Android.
```

File: src/stories/accidenz_commons.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+<div class="accidenz-commons has-text-black">
+  ${storyFn()}
+</div>`
+  ]
+ }}/>
+
+# Accidenz Commons
+Accidenz Commons,an Open-Licensed font is a contemporary, versatile, and neutral version of Akzidenz Grotesk was designed by Archetypo.
+
+It is used in Vocabulary in 3 font weights: regular (400), semibold (600) and bold (700).
```
Comment (dhruvkb) [inline src/stories/accidenz_commons.stories.mdx]: Just one: 400.

File: src/stories/accidenz_commons.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+  fontSizes={['2rem']}
+  fontFamily="Accidenz Commons"
+  fontWeight={700}
+  sampleText="0 1 2 3 4 5 6 7 8 9 + - * / % $" />
+
+<Typeset
+  fontSizes={['2rem']}
+  fontFamily="Accidenz Commons"
+  fontWeight={700}
+  sampleText=". ? ! ' & : ; @ # < > ( ) { } [ ]" />
```
Comment (dhruvkb) [inline src/stories/accidenz_commons.stories.mdx]: These variants do not exist. See `accidenz_commons.scss`.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
You cannot simply copy from Source Sans Pro to the other two files as they do not have the weights 600 and 700.

File: src/stories/roboto_condensed.stories.mdx (hunk 1 above)
Comment (megha070) [reply src/stories/roboto_condensed.stories.mdx]: Oh I am sorry, will make necessary changes

**Review by dhruvkb — APPROVED** [review]
LGTM 🚀

**Pull request merged**